Feature reference audit: distinct attributes (#987)
meilisearch/documentation: reference/features/distinct.md

# Distinct attribute

> A distinct attribute is a field whose value is shared by various documents. It is most commonly used to prevent MeiliSearch from returning several instances of similar documents.

Reviewer: ```suggestion
The distinct attribute is a special, user-designated field. It is most commonly used to prevent MeiliSearch from returning a set of several similar documents, instead forcing it to return only one.
```

> The value of a field configured as a distinct attribute will always be unique among returned documents. This means there will **never be more than one occurrence of the same value** of a distinct attribute field in the different documents returned by MeiliSearch.

Reviewer: ```suggestion
The value of a field configured as a distinct attribute will always be unique among returned documents. This means that among all the different documents returned by MeiliSearch, **there will never be more than one occurrence of the same value attached to the distinct attribute**.
```

Reviewer: This is a tricky concept to describe (but very easy to demonstrate). Not sure if what I've written is clear; feel free to take or leave it.

Author: Yeah, it's difficult to put it in words. I'm not sure either of our attempts make it much clearer, to be honest. 

```suggestion
The value of a field configured as a distinct attribute will always be unique among returned documents. This means **there will never be more than one occurrence of the same value** in the distinct attribute field among the returned documents.
```

> When multiple documents have the same value for a distinct attribute, MeiliSearch returns the first result after applying [ranking rules](/learn/core_concepts/relevancy.md#ranking-rules). If two or more documents are equivalent in terms of ranking, MeiliSearch returns the first result according to its `internal_id`.

Reviewer: ```suggestion
When multiple documents have the same value for the distinct attribute, MeiliSearch returns only the highest-ranked result after applying [ranking rules](/learn/core_concepts/relevancy.md#ranking-rules). If two or more documents are equivalent in terms of ranking, MeiliSearch returns the first result according to its `internal_id`.
```

### Example

> Suppose you have an e-commerce dataset. For an index that contains information about jackets, you may have several identical items in different variations such as color or size.

Reviewer: ```suggestion
Suppose you have an e-commerce dataset. For an index that contains information about jackets, you may have several identical items with minor variations such as color or size.
```

> As shown below, this dataset contains three documents for a Lee jeans leather jacket. One of the jackets is brown, one is black, and the last one is blue.

Reviewer: ```suggestion
As shown below, this dataset contains three documents representing different versions of a Lee jeans leather jacket. One of the jackets is brown, one is black, and the last one is blue.
```

]
```

> If a user searched for a `lee leather jacket`, MeiliSearch would return all three instances by default. This could cause the results to be cluttered with almost identical variations of the same item.

Reviewer: ```suggestion
By default, a search for `lee leather jacket` would return all three documents. This might not be desired, since displaying nearly identical variations of the same item can make results appear cluttered.
```

> In this case, you may want to ignore the different colors of a product. To do so, you can set `product_id` as a `distinctAttribute`.

Reviewer: ```suggestion
In this case, you may want to return only one document with the `product_id` corresponding to this Lee jeans leather jacket. To do so, you could set `product_id` as the `distinctAttribute`.
```

<CodeSamples id="distinct_attribute_guide_1" />

> By [setting `product_id` as a distinct attribute](/reference/api/distinct_attribute.md), search requests **will never return more than one jacket with the same `product_id`**.

Reviewer: ```suggestion
By setting `distinctAttribute` to `product_id`, search requests **will never return more than one document with the same `product_id`**.
```

> Once distinct attribute is configured, querying for `lee leather jacket` would only return the first document found. The response could look like this:

Reviewer: ```suggestion
After setting the distinct attribute as shown above, querying for `lee leather jacket` would only return the first document found. The response would look like this:
```
